### PR TITLE
Better Formatter tests

### PR DESCRIPTION
**Pull request by bigfoot90**
Writing tests this way can be harden, but you can be sure that your feature/fix for an formatter type will not impact on any other formatter type.

WDYT?
If you think that can be useful I'll continue to write tests for the remaining tests cases.

### CODE REVIEW

**Comment by nijel** [comment]
I think this is good idea, just make sure that all things currently covered will stay covered.

File: tests/Utils/FormatterTest.php [hunk 1]
```diff
@@ -230,6 +230,58 @@ public function mergeFormats()
...
+                    '&nbsp;&nbsp;&nbsp;&nbsp;*',
+            ),
+            'select with comment' => array(
+                'query' => 'select 1 # Comment',
+                'text' =>
+                    "SELECT\n" .
+                    "    1 # Comment" . "\n",
+                'cli' =>
+                    "\x1b[35mSELECT" . "\n" .
+                    "    \x1b[92m1 \x1b[37m# Comment\\x0A" . "\x1b[0m",
```
Comment (bigfoot90) [inline tests/Utils/FormatterTest.php]: Found first issue writing tests.
Why there is a `\x0A` in the comment?

Comment (bigfoot90) [inline tests/Utils/FormatterTest.php]: Fixing PR https://github.com/phpmyadmin/sql-parser/issues/123

**Comment by nijel** [comment]
Any progress on this?

**Comment by bigfoot90** [comment]
Yes, I was waiting for #123 to rebase onto.

**Comment by bigfoot90** [comment]
I'm ready for the review, yo can see that every test case in `testFormat` is covered by the new tests in `testFormat_new`.
Any comments or suggestions?

Actually only the `codacy/pr` tests are failing, this will be resolved when I remove the old tests in `testFormat` and replace `testFormat_new` with `testFormat`.

File: src/Utils/Formatter.php [hunk 2]
```diff
@@ -349,8 +349,10 @@ public function formatList($list)
             }
 
             // Checking if pointers were initialized.
-            /*
+            /**
              * Previous Token.
+             *
+             * @var Token $prev
```
Comment (nijel) [inline src/Utils/Formatter.php]: This is really not variable declaration, so the comment does not make much sense here...

File: src/Utils/Formatter.php [hunk 3]
```diff
@@ -349,10 +349,9 @@ public function formatList($list)
             }
 
             // Checking if pointers were initialized.
-            /*
-             * Previous Token.
-             */
             if ($prev !== null) {
+                /** @var Token $prev */
+
```
Comment (bigfoot90) [inline src/Utils/Formatter.php]: @nijel **WDYT** about this way?
I known that there is not a variable declaration, this way I say a bit different thig.
Here tha annotation just says that there is a variable `$prev` of type `Token`.

This annotation is demanded by IDE, else it says that there is not any property `value` on `null`.

If it is not ok I can revert this one, no problem.

Comment (nijel) [inline src/Utils/Formatter.php]: But $prev is declared before, I see no reason to have it annotated here...

Comment (bigfoot90) [inline src/Utils/Formatter.php]: There is already a annotation on the declaration *(on line 325)*, but PhpStorm is simply ignoring it.

**Comment by bigfoot90** [comment]
@nijel Removed `@var` anotation.
Are there other issues with this PR?
I can proceed to fix the method name `testFormat_new` and remove the old deprecated tests?

**Comment by nijel** [comment]
I think it's in good shape, please go ahead.

**Comment by nijel** [comment]
Thanks, I've merged it.

The coverage failure was caused by formatter handling differently `BEGIN` and `begin` as you've changed this. I've fixed this in 5fdfa5b0767c8b62e5dd16f1b99465c2d59c9d2f and 1a1168169c3d3acdf7ee3c5f1a839bcb66829297 and adjusted tests accordingly in 65fb611e091ef7d0b28039126733848bd20edb9d.

**Comment by bigfoot90** [comment]
Just to be clear. This way I'm testing that keywords are recognized and uppercased correctly.

**Comment by nijel** [comment]
They were uppercased properly, but the formatting didn't recognize because of being lowercase...

**Pull request merged**